test/server/test_adapter.py

```diff
@@ -114,9 +114,24 @@ def test_get_dataset(self):
 
         crn = self.warehouse_data[0]
 
-        self.backend.get_dataset(
+        crn_retrieved = self.backend.get_dataset(
             db_schema=crn.db_schema, db_table=crn.db_table, engine=crn.database.engine
         )
+
+        assert crn.db_columns == crn_retrieved.db_columns
+
+        cols: dict[str, list[SourceColumn]] = {}
+
+        # Indexing isn't used in the custom equality check
+        for col in crn.db_columns + crn_retrieved.db_columns:
+            if col.literal.name not in cols:
+                cols[col.literal.name] = [col]
+            else:
+                cols[col.literal.name].append(col)
+
+        for c1, c2 in cols.values():
+            assert c1.indexed == c2.indexed
+
```

Comment:
Nitpicking, but you could use a default dict of type list instead of the if condition